staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -311,6 +309,10 @@ func (s *SpdyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error)
 		return nil, err
 	}
 
+	ctx, cancel := context.WithCancel(req.Context())
+	defer utilnet.ExceedDeadlineOnCancel(ctx, conn)()
+	defer cancel()
```

Comment:
IIUC, the purpose of calling `conn.SetDeadline` was to unblock the `http.ReadResponse()` a few lines later when the context Done channel is closed.

Instead of introducing `ExceedDeadlineOnCancel`, can we use a `select` to achieve this? Something in this direction:

```golang
done := make(chan struct{})
go func() {
  resp, err = http.ReadResponse(responseReader, nil)
  close(done)
}()
select {
case <-done:
case <-ctx.Done():
  conn.Close() // this will unblock the http.ReadResponse
  return nil, fmt.Errorf(...)
}
...
```